tests/unit/snow/helpers/functional_test_files.py

```diff
@@ -0,0 +1,7 @@
+class FunctionalTestFile:
+    """A single test case with the required options"""
```

Comment:
not following how this file is used.

Reply:
All the test cases are passed as object of type FunctionalTestFile, instead of tuples.
In the test case we are using obj.databricks_sql, obj.source and obj.test_name.

It is used here(second last line):

```
 for part in parts[1:]:
                source, databricks_sql = part.split('-- databricks_sql:')
                source = source.strip().rstrip(';')
                databricks_sql = databricks_sql.strip().rstrip(';').replace('\\', '')
                test_name = file.replace('.sql', '')
                **suite.append(FunctionalTestFile(databricks_sql, source, test_name))**

return suite
```

Reply:
And in test case, its used like this:

`validate_source_transpile(databricks_sql=sample.databricks_sql, source={"snowflake": sample.source})`